File: LeetCode/Easy/0118-pascals-triangle/0118-pascals-triangle.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
        var rows = new int[numRows][];

        for (var i = 0; i < numRows; i++)
        {
            var row = new int[i + 1];
            row[0] = 1;
            row[i] = 1;
            for (var j = 1; j < i; j++)
            {
                row[j] = rows[i - 1][j - 1] + rows[i - 1][j];
            }
            rows[i] = row;
        }

        return rows;
    }
}

File: LeetCode/Easy/0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        triangle = [[1]]
        for row in range(1, numRows):
            left_shifted = [0] + triangle[row - 1]
            right_shifted = triangle[row - 1] + [0]

            triangle.append([a + b for a, b in zip(left_shifted, right_shifted, strict=True)])
        
        return triangle